### DISCUSSION TITLE
TemplateBinding is this possible?

### DISCUSSION THREAD

**Discussion by GirusVirus** [CommunityToolkit/Maui.Markup#202] [New Feature Discussions]
Hi, I just started with C# Markup and have ran into an issue with understanding how to create a custom reusable control. After creating a control template it all looks and works great, except I cannot seem to be able to bind to to the **BindableProperty**.  I looked everywhere for an example, but have not been able to find a way. Thanks,

**Comment by GirusVirus** [comment]
Hi, I was able to create a reusable control/view with the following code. However after much trial and error, I see that the "Bind" operator works with two different usages. see code. Now my question is, which one is correct or are they both acceptable? Is this the proper way to create a reusable control?


`public class HeaderView : ContentView
{
    public static readonly BindableProperty TextProperty =
        BindableProperty.Create(nameof(Text), typeof(string), typeof(HeaderView), string.Empty);


    public string Text
    {
        get => (string)GetValue(TextProperty);
        set => SetValue(TextProperty, value);
    }



    public HeaderView()
    {
        
        ControlTemplate = new ControlTemplate(() => new Grid
        {
            ColumnDefinitions = Columns.Define(Auto),

            Children =
            {
                new Border
                    {
                        Content = new Label()
                             //---------------------------------------------------------------------------                            
                            //.Bind(Label.TextProperty,static (HeaderView hv)=>hv.Text, source: this)
                            .Bind(Label.TextProperty, nameof(Text), source: this )
                            //----------------------------------------------------------------------------
                            .Margin(25, 4),

                        StrokeShape = new RoundRectangle { CornerRadius = new CornerRadius(10) },
                        StrokeThickness = .5
                    }
                    .Background(new SolidColorBrush(Color.FromArgb("#ffffffff")))
            }
        }.CenterHorizontal());


    }
}`

And usage would be like this:

`
Children =
{
     new HeaderView { Text = " Header View Title " },
     .
     .
}
`

**Reply by bijington** [reply]
Both are acceptable but as per our docs you want to use the former to avoid any reflection being used internally by .NET MAUI.

https://learn.microsoft.com/dotnet/communitytoolkit/maui/markup/extensions/bindable-object-extensions

**Reply by GirusVirus** [reply]
Thank you, I will do that.
